File: frontend/src/types/renter.ts
export interface Renter {
  id: string;
  full_name: string;
  phone?: string;
  email?: string;
  id_number?: string;
  current_unit_number?: string;
  current_property_name?: string;
  active_contract_id?: string;
  created_at: string;
}

export interface RenterDetail extends Renter {
  id_issued_date?: string;
  id_issued_place?: string;
  date_of_birth?: string;
  gender?: string;
  hometown?: string;
  occupation?: string;
  workplace?: string;
  emergency_contact_name?: string;
  emergency_contact_phone?: string;
  id_photo_links?: string[];
  notes?: string;
  contracts_history: RenterContractHistory[];
}

export interface RenterContractHistory {
  id: string;
  contract_number?: string;
  unit_number?: string;
  property_name?: string;
  start_date: string;
  end_date: string;
  status: string;
  monthly_rent: number;
}

export interface RenterCreate {
  full_name: string;
  phone?: string;
  email?: string;
  id_number?: string;
  id_issued_date?: string;
  id_issued_place?: string;
  date_of_birth?: string;
  gender?: string;
  hometown?: string;
  occupation?: string;
  workplace?: string;
  emergency_contact_name?: string;
  emergency_contact_phone?: string;
  id_photo_links?: string[];
  notes?: string;
}

export interface RenterUpdate {
  full_name?: string;
  phone?: string;
  email?: string;
  id_number?: string;
  id_issued_date?: string;
  id_issued_place?: string;
  date_of_birth?: string;
  gender?: string;
  hometown?: string;
  occupation?: string;
  workplace?: string;
  emergency_contact_name?: string;
  emergency_contact_phone?: string;
  id_photo_links?: string[];
  notes?: string;
}

export interface RenterFilter {
  search?: string;
  has_active_contract?: boolean;
}

File: backend/app/models/renter.py
from pydantic import BaseModel, Field
from typing import Optional
from datetime import datetime


class RenterCreate(BaseModel):
    full_name: str
    phone: Optional[str] = None
    email: str
    id_number: Optional[str] = None
    id_issued_date: Optional[str] = None
    id_issued_place: Optional[str] = None
    date_of_birth: Optional[str] = None
    gender: Optional[str] = None
    hometown: Optional[str] = None
    occupation: Optional[str] = None
    workplace: Optional[str] = None
    emergency_contact_name: Optional[str] = None
    emergency_contact_phone: Optional[str] = None
    id_photo_links: list[str] = Field(default_factory=list, max_length=5)
    notes: Optional[str] = None


class RenterUpdate(BaseModel):
    full_name: Optional[str] = None
    phone: Optional[str] = None
    email: Optional[str] = None
    id_number: Optional[str] = None
    id_issued_date: Optional[str] = None
    id_issued_place: Optional[str] = None
    date_of_birth: Optional[str] = None
    gender: Optional[str] = None
    hometown: Optional[str] = None
    occupation: Optional[str] = None
    workplace: Optional[str] = None
    emergency_contact_name: Optional[str] = None
    emergency_contact_phone: Optional[str] = None
    id_photo_links: Optional[list[str]] = None
    notes: Optional[str] = None


class RenterResponse(BaseModel):
    id: str
    full_name: str
    phone: Optional[str] = None
    email: Optional[str] = None
    id_number: Optional[str] = None
    current_unit_number: Optional[str] = None
    current_property_name: Optional[str] = None
    active_contract_id: Optional[str] = None
    created_at: Optional[str] = None


class RenterDetailResponse(RenterResponse):
    id_issued_date: Optional[str] = None
    id_issued_place: Optional[str] = None
    date_of_birth: Optional[str] = None
    gender: Optional[str] = None
    hometown: Optional[str] = None
    occupation: Optional[str] = None
    workplace: Optional[str] = None
    emergency_contact_name: Optional[str] = None
    emergency_contact_phone: Optional[str] = None
    id_photo_links: list[str] = Field(default_factory=list)
    notes: Optional[str] = None
    contracts_history: list = []
